ynput/OpenPype
# Max: updated implementation of save_scene + small QOL improvements to host

## Changelog Description
- Removed `has_unsaved_changes` from Max host as it looks to have been unused and unimplemented.
- Added and implemented `workfile_has_unsaved_changes` to Max host.
- Mirrored the Houdini host to implement the above into `save_scene` publish for Max.
- Added a line to `startup.ms` which opens the usual 'default' menu inside of Max (see screenshots).


Current (Likely opens this menu due to one or more of the startup scripts used to insert OP menu):
<img width="188" alt="image" src="https://github.com/ynput/OpenPype/assets/20257044/263cd939-b64d-4bbe-b02d-60cfa6e9ac80">

New:
<img width="179" alt="image" src="https://github.com/ynput/OpenPype/assets/20257044/86077798-36fc-40f1-b50a-bf3d0956b2cb">


## Additional info
It's my understanding that the `has_unsaved_changes` is now legacy so I have replaced and implemented it using `workfile_has_unsaved_changes`. 

The change to the `startup.ms` is really just QOL for artists as the view it currently defaults to is not useful nor is it the usual default.


## Testing notes:
1. Start 3dsmax 
2. Check that on the right hand side it has defaulted to the 'create' view/menu/mode
3. Open the 'Publish/Create' OpenPype Menu
4. Create new 'workfile' publish
5. Publish 
6. Check the log of `save_scene` is NOT saving if no changes to scene were made; or is saving when changes have been made (it should display INFO or DEBUG information in either case).
